stackslib/src/net/api/getsortition.rs

```diff
@@ -0,0 +1,380 @@
+        } else {
+            value.1.as_str()
+        };
+        match value.0.as_str() {
+            "consensus" => Ok(Self::ConsensusHash(
+                ConsensusHash::from_hex(hex_str).map_err(|e| Error::DecodeError(e.to_string()))?,
+            )),
+            "burn" => Ok(Self::BurnchainHeaderHash(
+                BurnchainHeaderHash::from_hex(hex_str)
+                    .map_err(|e| Error::DecodeError(e.to_string()))?,
+            )),
+            "burn_height" => Ok(Self::BlockHeight(
+                value
+                    .1
+                    .parse::<u64>()
+                    .map_err(|e| Error::DecodeError(e.to_string()))?,
+            )),
+            other => Err(Error::DecodeError(format!("Unknown query param: {other}"))),
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct GetSortitionHandler {
+    pub query: QuerySpecifier,
+}
+
+impl GetSortitionHandler {
+    pub fn new() -> Self {
+        Self {
+            query: QuerySpecifier::Latest,
+        }
+    }
+}
+/// Decode the HTTP request
+impl HttpRequest for GetSortitionHandler {
+    fn verb(&self) -> &'static str {
+        "GET"
+    }
+
+    fn path_regex(&self) -> Regex {
+        Regex::new(&format!("^{RPC_SORTITION_INFO_PATH}$")).unwrap()
+    }
+
+    /// Try to decode this request.
+    /// There's nothing to load here, so just make sure the request is well-formed.
+    fn try_parse_request(
+        &mut self,
+        preamble: &HttpRequestPreamble,
+        _captures: &Captures,
+        query: Option<&str>,
+        _body: &[u8],
+    ) -> Result<HttpRequestContents, Error> {
+        if preamble.get_content_length() != 0 {
+            return Err(Error::DecodeError(
+                "Invalid Http request: expected 0-length body".to_string(),
+            ));
+        }
+
+        let req_contents = HttpRequestContents::new().query_string(query);
+        if req_contents.get_query_args().len() > 1 {
+            return Err(Error::DecodeError(
+                "May only supply up to one query argument".into(),
+            ));
+        }
+        self.query = QuerySpecifier::Latest;
+        for (key, value) in req_contents.get_query_args().iter() {
+            self.query = QuerySpecifier::try_from((key, value))?;
+        }
+
+        Ok(req_contents)
+    }
+
+    fn metrics_identifier(&self) -> &str {
+        RPC_SORTITION_INFO_PATH
+    }
+}
+
+impl RPCRequestHandler for GetSortitionHandler {
+    /// Reset internal state
+    fn restart(&mut self) {
+        self.query = QuerySpecifier::Latest;
+    }
+
+    /// Make the response
+    fn try_handle_request(
+        &mut self,
+        preamble: HttpRequestPreamble,
+        _contents: HttpRequestContents,
+        node: &mut StacksNodeState,
+    ) -> Result<(HttpResponsePreamble, HttpResponseContents), NetError> {
+        let result =
+            node.with_node_state(|_network, sortdb, _chainstate, _mempool, _rpc_args| {
+                let query_result = match self.query {
```

Comment:
An optimization you can make: the `_network` struct has `.burnchain_tip`, which is the canonical block snapshot.  If any of these query specifiers match `.burnchain_tip`'s corresponding fields, then you can just return `_network.burnchain_tip` instead of doing a DB load. I suspect that this will be the case the majority of the time.